app/controllers/wishlist_controller.rb

```diff
@@ -1,6 +1,57 @@
 class WishlistController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_wishlist_item, except: [:index]
 
   # GET /wishlist
   def index
+    @wishlist = current_user.wishlist.includes(course: :course_instances)
+  end
+
+  # POST /wishlist
+  def add_course
+    respond_to do |format|
+      if @wishlist_item.save
+        flash[:notice] = 'Course successfully saved to your wishlist'
+        format.html { redirect_to params[:url] || wishlist_path }
+      else
+        flash[:notice] = 'An error occurred adding the course to your wishlist'
+        format.html { redirect_to params[:url] || wishlist_path }
+      end
+    end
+  end
+
+  # PUT /wishlist
+  def update
+    respond_to do |format|
+      if @wishlist_item.update(wishlist_item_params)
+        format.html { redirect_to wishlist_path }
+      else
+        format.html { redirect_to wishlist_path }
```

Comment:
I think that's leftover from me thinking about if I wanted to flash an error message in case of failure to update, but then I never actually ended up doing that.